main.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"unicode"
+)
+
+func IsSep(r rune) bool { return unicode.IsSpace(r) || unicode.IsPunct(r) || unicode.IsSymbol(r) }
+
+func wordOccurrences(s string) (words []string, occurrences []uint) {
+	var w []string
+	var tmp []rune
+	wasSep := true
+	for _, r := range s {
+		if IsSep(r) {
+			if !wasSep {
+				w = append(w, string(tmp))
+				tmp = nil
+			}
+			wasSep = true
+		} else {
+			wasSep = false
+			tmp = append(tmp, r)
+		}
+	}
```

Comment:
This fragment looks similar to the function `words` (a function that splits a string into words), which wrote in the class. If this is what you want, why wouldn't you call it explicitly?